weather.py

```diff
@@ -1,72 +1,143 @@
+
+@dataclass
+class Temp:
+    Value: float
+    Unit: str
+    UnitType: int
+
+@dataclass
+class Temperature:
+    Metric: Temp
+    Imperial: Temp
+    
+@dataclass
+class Weather:
+    LocalObservationDateTime: str
+    EpochTime: int
+    WeatherText: str
+    WeatherIcon: int
+#    PrecipitationType: str
+    HasPrecipitation: bool
+    IsDayTime: bool
+    Temperature: Temperature
+    MobileLink: str
+    Link: str
+
+#data = {'LocalObservationDateTime': '2023-01-24T20:03:00+00:00', 'EpochTime': 1674590580, 'WeatherText': 'Mostly cloudy', 'WeatherIcon': 38, 'HasPrecipitation': False, 'PrecipitationType': None, 'IsDayTime': False, 'Temperature': {'Metric': {'Value': 6.7, 'Unit': 'C', 'UnitType': 17}, 'Imperial': {'Value': 44.0, 'Unit': 'F', 'UnitType': 18}}, 'MobileLink': 'http://www.accuweather.com/en/gb/leeds/ls1-6/current-weather/712327?lang=en-us', 'Link': 'http://www.accuweather.com/en/gb/leeds/ls1-6/current-weather/712327?lang=en-us'}
+
 def run():
-    # Define the api endpoint.
     api_endpoint = 'https://dataservice.accuweather.com'
 
-    # Define our api key. This is sensitive data, so we should consider passing
-    # it in as a flag, or reading from the environment.
-    apikey = 'bTWRzyxjP94K3Wn66NlQsr3h2qppp3cs'
-
-    # Define the api path for getting locations.
-    locations_path = 'locations/v1/regions'
-
-    # Here we call the get_locations() function passing in the locations url
-    # and api key.
-    locations = get_locations('{}/{}?apikey={}'.format(
-        api_endpoint, locations_path, apikey))
-
-    # ...and for example, let's say we want to know Africa's ID...
-    print(locations['Africa'])
-    # or Europe for that matter...
-    print(locations['Europe'])
-    # ...and let's pretend the user specifed the value as a command-line
-    # argument...
-    location = 'Central America'
-    print(locations[location])
-    # worth noting that what if the user specifed the location in lower-case?
-    # we could add all the locations to the dict in lower-case, then convert
-    # the user specifed location to lower-case too.
-    # Starting to sound like how to store stuff in a database. No coincidence!
+    apikey = '4bAzYrckrKgStAXpCiekZLyvkAGkHLmG'
+
+    # If user uses lower case,  we could add all the locations to the dict in lower-case, 
+    # then convert
+    
+    # Location to get the weather for
+    city = 'York'
+    # Admin area
+    admin_areas_path = 'locations/v1/adminareas'
+
+    admin_areas = get_locations('{}/{}?apikey={}'.format(
+        api_endpoint, admin_areas_path, apikey))
+
+    admin_area_code = (admin_areas[(city)])
```

Comment:
Why the parentheses around `admin_areas[(city)]`? In fact you don't need them around `city` either.

Reply:
I'm not sure. I was just trying to get it working and this worked. I've updated as suggested though.